### PR TITLE
F3DLoader rework

### PR DESCRIPTION
**Pull request by mwestphal**
- Separate F3DLoader in F3DLoader and F3DStarter for #52 
- Remove all mention of F3DOptions from F3DLoader
- Make --no-render F3DStarter specifc and not an actual option
- Separate F3DLoader::loadFile in multiple methods
- Improve testing for right/left key

### CODE REVIEW

File: src/library/F3DLoader.h [hunk 1]
```diff
@@ -11,23 +11,26 @@
...
 {
 public:
   enum LoadFileEnum
   {
+    LOAD_FIRST = -2,
     LOAD_PREVIOUS = -1,
     LOAD_CURRENT = 0,
-    LOAD_NEXT = 1
+    LOAD_NEXT = 1,
+    LOAD_LAST = 2
```
Comment (mwestphal) [inline src/library/F3DLoader.h:31]: not sure about that, maybe I should use int min and int max.

File: src/library/F3DNSDelegate.mm [hunk 2]
```diff
@@ -2,37 +2,32 @@
...
-@synthesize RenderWindow;
+@synthesize Starter;
 
 // ----------------------------------------------------------------------------
 - (BOOL)application:(NSApplication*)theApplication openFile:(NSString*)filename
 {
   (void)theApplication;
-  Loader->AddFile([filename UTF8String]);
-  Loader->LoadFile();
-  RenderWindow->Render();
```
Comment (mwestphal) [inline src/library/F3DNSDelegate.mm:24]: not sure about this removal, but I do not see why it should be necessary.

**Comment by mwestphal** [comment]
a

**Comment by mwestphal** [comment]
image

File: src/library/F3DNSDelegate.mm (hunk 2 above)
Comment (mwestphal) [reply src/library/F3DNSDelegate.mm:24]: to test on macos

File: src/library/F3DOptions.cxx [hunk 3]
```diff
@@ -190,7 +190,7 @@ F3DOptions ConfigurationOptions::GetOptionsFromArgs(std::vector<std::string>& in
     this->DeclareOption(grp1, "version", "", "Print version details");
     this->DeclareOption(grp1, "readers-list", "", "Print the list of file types");
     this->DeclareOption(grp1, "verbose", "", "Enable verbose mode, providing more information about the loaded data in the console output", options.Verbose);
-    this->DeclareOption(grp1, "no-render", "", "Verbose mode without any rendering, only for the first file", options.NoRender);
+    this->DeclareOption(grp1, "no-render", "", "Verbose mode without any rendering, only for the first file", options.NoRender, true, false);
```
[the remark this answers is not in the document]
Comment (mwestphal) [reply src/library/F3DOptions.cxx:193]: out of scope of this PR

File: src/library/F3DNSDelegate.mm (hunk 2 above)
Comment (mwestphal) [reply src/library/F3DNSDelegate.mm:24]: please test: https://github.com/f3d-app/f3d/actions/runs/1705506643

**Review by Meakk — APPROVED** [review]

**Pull request merged**